firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -281,4 +320,11 @@ void recalculateAndSaveOverlays(Set<DocumentKey> documentKeys) {
 
     return results;
   }
+
+  /** Returns a base document that can be used to apply `overlay`. */
```

Comment:
optional reword: Returns the base document that corresponds to the provided overlay.

Reply:
Skipped as "corresponds" doesn't sound correct when we return an invalid document for a set() even though there is actually a valid document in the cache.